text/0000-override-env-by-default.md

```diff
@@ -0,0 +1,54 @@
+# Meta
+[meta]: #meta
+- Name: Override Env Vars by Default
+- Start Date: 2020-07-26
+- Author(s): @sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This RFC proposes that we change the behavior of `/env/`, `/env.build/`, and `/env.launch/` directories such that the default, suffix-less behavior is equivalent to `VAR.override` instead of `VAR.append`+`VAR.delim=:`.
+
+# Motivation
+[motivation]: #motivation
+
+The current behavior is surprising, especially because adding `.append` silently removes the delimiter. Changing to `.override` and requiring a `.delim` file to add a delimiter would be more clear.
+
+Additionally, not all environment variables are paths, so assuming `:` may also surprise users when two unrelated buildpacks set the same env var without an extension.
```

Comment:
Additional reasons for this change... 

- Even when an environment variable contains a path that needs to be joined, there's no guarantee that `.append` is the correct choice over `.prepend`. Making one the default hides the need to think about what is the most appropriate choice, and results in asymmetrical syntax for the two use-cases.
- Buildpack authors have to remember which of `.append` vs `.prepend` is the default, or else continually look in the docs. This seems less likely to occur if the default were `override`.
- Whilst many env vars in classic buildpacks are paths, my understanding is that for CNBs the platform will automatically set env vars for many common directories (https://github.com/buildpacks/spec/blob/18d50ea3d637d4fdcfe77b4a88a17e8a19b80799/platform.md#buildpack-environment) reducing the amount of path handling that is required by the buildpack

Reply:
Mind proposing these as a change via the github UI?